packages/pluggableWidgets/datagrid-web/src/components/Table.tsx

```diff
@@ -94,50 +96,56 @@ export function Table<T>(props: TableProps<T>): ReactElement {
     );
     const tableColumns: Array<ColumnWithStrictAccessor<{ item: T }>> = useMemo(
         () =>
-            props.columns.map((column, index) => ({
-                id: index.toString(),
-                accessor: "item",
-                Header: typeof column.header === "object" ? column.header.value : column.header,
-                filter: "text",
-                hidden: column.hidable === "hidden",
-                canHide: column.hidable !== "no",
-                canDrag: column.draggable,
-                customFilter:
-                    column.filterable === "custom"
-                        ? props.filterRenderer((children: ReactNode) => <div className="filter">{children}</div>, index)
-                        : null,
-                disableSortBy: !column.sortable,
-                disableResizing: !column.resizable,
-                disableFilters: column.filterable === "no",
-                sortType: (rowA: Row<{ item: T }>, rowB: Row<{ item: T }>, columnId: IdType<object>): number => {
-                    const valueA = props.valueForSort(rowA.values[columnId], Number(columnId)) || "";
-                    const valueB = props.valueForSort(rowB.values[columnId], Number(columnId)) || "";
-                    // Values should always be sorted in ASC mode https://github.com/tannerlinsley/react-table/pull/2504
-                    if (typeof valueA === "string" && typeof valueB === "string") {
-                        return valueA.localeCompare(valueB);
-                    } else if (typeof valueA === "boolean" && typeof valueB === "boolean") {
-                        // True first
-                        return valueA === valueB ? 0 : valueA ? -1 : 1;
-                    } else if (valueA instanceof Date && valueB instanceof Date) {
-                        return (valueA as Date).getTime() - (valueB as Date).getTime();
-                    }
-                    return Number(valueA) - Number(valueB);
-                },
-                Cell: ({ cell, value }) =>
-                    props.cellRenderer(
-                        (children: ReactNode) => (
-                            <div
-                                {...cell.getCellProps()}
-                                {...(!props.columnsResizable ? { style: { flex: "1 1 0px" } } : {})}
-                                className="td"
-                            >
-                                {children}
-                            </div>
+            props.columns.map((column, index) => {
+                const weight = getWeight(column.width, column.size ?? 1);
+                const totalWeight = props.columns.reduce((p, c) => p + getWeight(c.width, c.size ?? 0), 0);
```

Comment:
You compute it for every column, which feels a bit wasteful

Reply:
Also why defaults for size here and one line above are different? I suspect the answer, but imho it is quite spooky difference => the question is whether we can make it a bit more explicit

Reply:
I moved code to analyze the visible columns directly into the header code, so we just have weight being computed for each column